applications/callflow/src/cf_endpoint.erl

```diff
@@ -529,12 +546,34 @@ maybe_owner_called_self(Endpoint, Properties, Call) ->
         'false' ->
             lager:info("owner ~s stop calling your self...stop calling your self...", [OwnerId]),
             {'error', 'owner_called_self'}
+    end;
+maybe_owner_called_self(Endpoint, Properties, <<"sms">>, Call) ->
+    AccountId = whapps_call:account_id(Call),
+    DefTextSelf = whapps_account_config:get_global(AccountId, ?CF_CONFIG_CAT, <<"default_can_text_self">>, 'true'),
+    CanCallSelf = wh_json:is_true(<<"can_text_self">>, Properties, DefTextSelf),
+    EndpointOwnerId = wh_json:get_value(<<"owner_id">>, Endpoint),
+    OwnerId = whapps_call:kvs_fetch('owner_id', Call),
```

Comment:
Should probably use `whapps_call:owner_id(Call)` since we want the caller's owner ID, if any.